linode/manager
# upcoming: [DI-21694] - Added Resource Select component to the Create-alert form

## Description 📝

- Added the Resource Multi Select component under General Information section to the Create alert form.

## Changes  🔄
- Added a MultiSelect Autocomplete component for selecting Resources
- Added Unit Test for the Resource Multi Select component
- Changed few variables and properties to camelCase

## Target release date 🗓️

Please specify a release date (and environment, if applicable) to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-11-19 at 21 38 14](https://github.com/user-attachments/assets/5c482b69-527e-4848-8cf9-fbcc2c5531ec) | ![Screenshot 2024-11-26 at 18 43 55](https://github.com/user-attachments/assets/fda80b26-7823-409c-ac37-48ca3cab8f4e) |

## How to test 🧪

### Prerequisites

(How to setup test environment)

- The tabs are controlled by a featureFlag called aclpAlerting, the flag has been currently disabled. For testing enable the definitions part of the aclpAlerting flag to be true.
- The API endpoints are not live yet, so use Legacy MSW Handlers for the mock responses.

### Verification steps

(How to verify changes)

- In Monitor (once the prerequisites are followed) , Alerts tab should be visible next to Dashboards.
- Under that,  Definition tab and a Create button should be visible.
- On clicking Create, the Form Page should be visible.
- The label of Resources component should change to Clusters if Service is `Database `

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅
